test/basic.js

```diff
@@ -84,6 +84,31 @@ tape('verify', function (t) {
   })
 })
 
+tape('commend', function (t) {
+  t.plan(6)
+
+  var feed = create()
+  var evilfeed = create(feed.key, {secretKey: feed.secretKey})
+
+  feed.append('test', function (err) {
+    t.error(err, 'no error')
+
+    evilfeed.append('t\0st', function (err) {
+      t.error(err, 'no error')
+
+      feed.commend(0, function (err, roots) {
+        t.error(err, 'no error')
+        t.ok(roots instanceof Array)
+      })
+
+      evilfeed.commend(0, function (err, roots) {
+        t.ok(err === null, 'no error')
+        t.ok(roots instanceof Array)
```

Comment:
This is good test coverage.  A nitpick would be that the test doesn't check that evilfeed and feed have different root hashes (the point of getting root hashes is to ensure matching content is the same and differing content is different without having to compare all the content).

It would be good to check evilfeed after the roots of feed are returned, and assert that the two have the same number of hashes but that those hashes are different.

Reply:
:+1: on @xloem comments